moto/emr/responses.py

```diff
@@ -989,3 +1113,120 @@ def terminate_job_flows(self):
     <RequestId>2690d7eb-ed86-11dd-9877-6fad448a8419</RequestId>
   </ResponseMetadata>
 </TerminateJobFlowsResponse>"""
+
+PUT_AUTO_SCALING_POLICY = """<PutAutoScalingPolicyResponse xmlns="http://elasticmapreduce.amazonaws.com/doc/2009-03-31">
+  <PutAutoScalingPolicyResult>
+    <ClusterId>{{cluster_id}}</ClusterId>
+    <InstanceGroupId>{{instance_group.id}}</InstanceGroupId>
+    {% if instance_group.auto_scaling_policy is not none %}
```

Comment:
This block is a duplicate of one in ADD_INSTANCE_GROUP_RESPONSE.  Are there any patterns established in moto for re-use of common templated blocks?

Reply:
As far as I'm aware, there's no existing pattern. But with a block this size, it would be good to only declare it once.

The best approach in terms of extendability is, IMO, to change the `response_template()` signature in moto/core/responses.py, and allow for inline templates. Something like this:  
```
def response_template(self, source, inline_templates = {})
  # inline_templates = {"AUTOSCALING_POLICY": "AUTOSCALING_POLICY_TEMPLATE"}
  ..
  template = self.environment.get_template(template_id)
  for k, v in inline_templates:
    template.substibute(k, self.environment.get_template(v))
  return template
```

Reply:
I'm going with something similar, but which enables the use of jinja2's {% include 'INLINE_TEMPLATE_NAME' %}, by dropping the inline templates into the loader dict.  To make inline templates resolveable by their name, they get stored in the loader dict keyed with their name, which is different than for top-level templates (keyed by object id), but not harmful in my opinion.  Let me know if you disagree.

Reply:
Done.